### PR TITLE
[frontend] fix bug: default user location;

### PR DESCRIPTION
**Pull request by jerryfreshworks**
- showing default user location;

- `Get /booking` with default user location

- Add notification alert when save the user default location.

The reason of the bug:  material-ui component`Autocomplete`, can't set async value to `defaultValue` props.

### CODE REVIEW

**Review by fwpushan — APPROVED** [review]
Great job, good learning for me thanks! 👍

File: frontend/src/components/form/BookingsSearch.tsx [hunk 1]
```diff
@@ -23,8 +23,12 @@ import {
   getUserLocation,
   updateUserLocation,
 } from 'util/apiHelper';
-import { AutoCompleteField, ACFC } from './inputs/AutoCompleteField';
+import {
+  AutoCompleteField,
+  ACFC,
+} from 'components/form/inputs/AutoCompleteField';
```
Comment (michel-fresh) [inline frontend/src/components/form/BookingsSearch.tsx:29]: 💯

File: frontend/src/components/form/BookingsSearch.tsx [hunk 2]
```diff
@@ -36,20 +40,37 @@ export default function Search({
...
+        getSearchResults({
+          url: '/booking',
+          method: 'GET',
+          params: { locationId: fetchedUserLocation?.id },
+        });
+      } else {
+        setUserLocation(null);
+      }
     }
     fetchLocation();
```
Comment (michel-fresh) [inline frontend/src/components/form/BookingsSearch.tsx:64]: also option to use IIFE (async () => {})()

File: frontend/src/components/form/BookingsSearch.tsx [hunk 3]
```diff
@@ -36,20 +40,37 @@ export default function Search({
...
   }, [setLocations, setUserLocation]);
+
+  const { addAlert } = useAlert();
+
   return (
     <Box>
       <Formik
-        initialValues={Initial}
+        initialValues={{ ...Initial, locationId: userLocation?.id }}
+        enableReinitialize
```
Comment (michel-fresh) [inline frontend/src/components/form/BookingsSearch.tsx:73]: not sure why this isn't on by default in the formik library

File: frontend/src/components/form/BookingsSearch.tsx [hunk 4]
```diff
@@ -84,8 +107,16 @@ export default function Search({
                   color="primary"
                   size="small"
                   startIcon={<SaveIcon />}
-                  onClick={() => {
-                    updateUserLocation(values.locationId);
+                  onClick={async () => {
+                    try {
+                      await updateUserLocation(values.locationId);
+                      addAlert('Update User Default Location Successfully');
```
Comment (michel-fresh) [inline frontend/src/components/form/BookingsSearch.tsx:113]: I feel like the term "User" shouldn't be user facing, it's kind of impersonal. In this context, I'd just say "Your preferred location was saved"

**Review by michel-fresh — APPROVED** [review]
sweet

File: frontend/src/components/form/BookingsSearch.tsx (hunk 4 above)
Comment (jerryfreshworks) [reply frontend/src/components/form/BookingsSearch.tsx:113]: Thanks always, I'm very struggling with correct sentences.

File: frontend/src/components/form/BookingsSearch.tsx (hunk 2 above)
Comment (jerryfreshworks) [reply frontend/src/components/form/BookingsSearch.tsx:64]: cool!

File: frontend/src/components/form/BookingsSearch.tsx (hunk 3 above)
Comment (jerryfreshworks) [reply frontend/src/components/form/BookingsSearch.tsx:72]: ha, that's true.

**Pull request merged**